Update release process document (#341)
Babylonpartners/ios-playbook: Cookbook/Technical-Documents/ReleaseProcess.md

*It starts after the App Center build has been delivered and it can take several cycles*

1. Testers will then begin their work against the build you just created.
1. Any hotfix should target the release branch first.

> 1. These PRs need to be reviewed by the relevant squad or platform QA (Bibin Paul) & one of the release engineers assigned to the release. This is to ensure visibility of changes being requested to be made, and to ensure the correct builds are available for validation.

Reviewer: Being the process we should just reference the role and not the name. If we want to keep the name then we can have a separate section where we can assign people to the roles (*release manager, platform QA, etc...)
```suggestion
1. These PRs need to be reviewed by the relevant squad or **platform QA** & one of the release engineers assigned to the release. This is to ensure visibility of changes being requested to be made, and to ensure the correct builds are available for validation.
```

   * Attach the zipped `xcarchive` as an artefact to the GitHub release (if you're using the automated release command, you can find the `*.xcarchive.zip` in the Artifacts top section in the CI build).
1. Merge `release` branch back to `develop`:
	* Open the Release PR ( PR from `release` branch targeting `develop`) which has been automatically created.
  * Resolve the conflicts (if any).

> 	* In case there are any changes besides updates to app and build versions, then assign as reviewers all the engineers who worked on those changes and remove the ones automatically assigned by PullAssigners. After merging with `develop` the extra changes may be there due to either resolving conflict or because hot fix PR wasn't merged to `develop`.

Reviewer 2: ```suggestion
	* In case there are changes other than updates to app and build versions, the release engineer should assign as reviewers all the engineers who worked on those changes and remove the ones automatically assigned by PullAssigners. There might be conflicts when merging the extra changes with `develop` or in the case of a hot fix PR not been merged previously to `develop`.
```

Author: > There might be **conflicts** when merging the extra changes with `develop` or in the case of a hot fix PR not been merged previously to `develop`.

I was more trying to describe scenarios in which we have changes. It's very rare, since everything is supposed to be merged to `develop` soon after it gets to `release` branch. Maybe we can add even more details here to make it clear. See the following, what do you think?

- In case there are changes other than updates to app and build versions after merging the changes from `develop`, the release engineer should assign as reviewers all the engineers who worked on those changes and remove the ones automatically assigned by PullAssigners. In current workflow every change integrated to release branch is supposed to be go into to `develop` soon after, so differences should be minimal. Any differences might be a result of resolving conflicts or a hot fix PR not being merged previously to `develop`.

Reviewer 2: Sounds good 😃

> 	* In case there are any changes besides updates to app and build versions, then assign as reviewers all the engineers who worked on those changes and remove the ones automatically assigned by PullAssigners. After merging with `develop` the extra changes may be there due to either resolving conflict or because hot fix PR wasn't merged to `develop`.

Author: ```suggestion
	*  In case there are changes other than updates to app and build versions after merging the changes from `develop`, the release engineer should assign as reviewers all the engineers who worked on those changes and remove the ones automatically assigned by PullAssigners. In current workflow every change integrated to release branch is supposed to be go into to `develop` soon after, so differences should be minimal. Any differences might be a result of resolving conflicts or a hot fix PR not being merged previously to `develop`.
```

	* Set the _Merge_ label once all the required reviewers have approved it.
1. Update the [release calendar](#release-calendar)
	* Tip: You can use the following command in your Terminal to list the tickets and authors who participated in the Release, to ask them the time they took for each ticket
      ```

>       git log --format="%<(25)%an | %s" origin/develop..origin/release/babylon/4.15.0 | grep -vE "^(Steve|iOS Bot) *\|" | sort

Reviewer 3: I'm thinking we should use a placeholder instead of hardcoded example value `4.15.0`, to avoid people copy/pasting that command and not realizing they need to update the branch name

```suggestion
      git log --format="%<(25)%an | %s" origin/develop..origin/release/{flavor}/{version} | grep -vE "^(Steve|iOS Bot) *\|" | sort
```
